lib/internal/util/comparisons.js

```diff
@@ -107,8 +108,9 @@ function strictDeepEqual(val1, val2, memos) {
     if (val1.length !== val2.length) {
       return false;
     }
-    const keys1 = getOwnNonIndexProperties(val1);
-    if (keys1.length !== getOwnNonIndexProperties(val2).length) {
+    const keys1 = getOwnNonIndexProperties(val1, ONLY_ENUMERABLE);
+    const keys2 = getOwnNonIndexProperties(val2, ONLY_ENUMERABLE);
+    if (keys1.length !== keys2.length) {
```

Comment:
The reason for needing `getOwnNonIndexProperties` should be brought up to the TC39 _(if it hasn't already)_. It'd be good to let them know that Node is having to use engine level APIs to work around a JS lang limitation _(no way to return a value with methods that return arrays of keys)_.